# Question 11
-----

In [1]:
from scipy.integrate import quad
import math

class normal:
    def __init__(self, u, sigma):
        self.u = u
        self.sigma = sigma
        self.root2pi = math.sqrt((2 * math.pi)) 
    
    def normal_func(self, x):
        return (1 / (self.sigma * self.root2pi)) * math.exp((-1 * pow((x - self.u), 2)) / (2 * pow(self.sigma, 2)))

    def integrate(self, lower_limit, upper_limit):
        result, error = quad(self.normal_func, lower_limit, upper_limit)
        
        return result

    def locate(self, area):
        fx = 0
        x = -6
        while fx < area:
            fx = self.integrate(-6, x)
            x = x + 0.00001
            
        return x

In [2]:
z = normal(0, 1)
loc_1 = z.locate(0.45)
loc_2 = z.locate(0.9)
print(loc_1)
print(loc_2)

-0.12565000005796295
1.2815699999419503


$$0.13\sigma + 240 = \mu$$

$$250 - 1.28\sigma = \mu$$

$$0.13\sigma + 240 = 250 - 1.28\sigma$$

$$1.41\sigma = 10$$

$$\sigma = 7.09$$

$$\mu = (0.13 * 7.09) + 240$$

$$\mu = 240.9217$$

In [3]:
dist = normal(240.9217, 7.09)
prob = dist.integrate(0, 235)
print("P(X < 235) = " + str(prob))

P(X < 235) = 0.20179731448734936
